app/main.py

```diff
@@ -130,7 +130,7 @@ async def on_startup() -> None:
 
 @app.on_event("shutdown")
 async def on_shutdown() -> None:
-    await close_caches()
+    close_caches()
```

Comment:
```suggestion
    try:
        await close_caches()
     except TypeError:
         pass
```

Reply:
After another look, it seems the await exception is raised in the function itself.

We should try to close the caches and just except and pass the returned TypeError IMO

Reply:
addressed